src/Features/LanguageServer/Microsoft.CodeAnalysis.LanguageServer/HostWorkspace/ExtensionManager.cs

```diff
@@ -2,40 +2,20 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
-using System;
-using System.Collections.Generic;
 using System.Composition;
-using System.Linq;
-using System.Text;
-using System.Threading.Tasks;
 using Microsoft.CodeAnalysis.Extensions;
 using Microsoft.CodeAnalysis.Host.Mef;
 using Microsoft.Extensions.Logging;
 
 namespace Microsoft.CodeAnalysis.LanguageServer.HostWorkspace;
 
 [ExportWorkspaceService(typeof(IExtensionManager), ServiceLayer.Host), Shared]
-internal class ExtensionManager : IExtensionManager
+[method: ImportingConstructor]
+[method: Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+internal sealed class ExtensionManager(ILoggerFactory loggerFactory) : AbstractExtensionManager
 {
-    private readonly ILogger _logger;
+    private readonly ILogger _logger = loggerFactory.CreateLogger(nameof(ExtensionManager));
 
-    [ImportingConstructor]
-    [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
-    public ExtensionManager(ILoggerFactory loggerFactory)
-    {
-        _logger = loggerFactory.CreateLogger(nameof(ExtensionManager));
-    }
-
-    public bool CanHandleException(object provider, Exception exception) => true;
-
-    public void HandleException(object provider, Exception exception)
-    {
-        _logger.Log(LogLevel.Error, exception, $"{provider.GetType().ToString()} threw an exception.");
-    }
-
-    public bool IsDisabled(object provider)
-    {
-        // We don't have an UI to allow disabling yet
-        return false;
-    }
+    protected override void HandleNonCancellationException(object provider, Exception exception)
+        => _logger.Log(LogLevel.Error, exception, $"{provider.GetType()} threw an exception.");
```

Comment:
@dibarbet Despite having no UI for ut, we're still not *disabling* the providers when they crash.  IMO, we should do that in LSP, otherwise we can just continue calling into them, and they will continue crashing.  Thoughts?

Reply:
Talked offline - fine with this for this PR.  Tracking issue here https://github.com/dotnet/roslyn/issues/70844 to enable a similar UI that VS has.